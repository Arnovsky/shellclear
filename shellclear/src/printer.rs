use crate::data::FindingSensitiveCommands;
use prettytable::{Cell, Row, Table};

pub fn show_sensitive_findings(findings: Vec<FindingSensitiveCommands>) {
    let mut table = Table::new();

    table.add_row(Row::new(vec![
        Cell::new("Shell"),
        Cell::new("Name"),
        Cell::new("Command"),
    ]));

    for f in findings {
        let cells = vec![
            Cell::new(&format!("{:?}", f.shell_type)),
            Cell::new(
                f.finding
                    .iter()
                    .map(|f| f.name.to_owned())
                    .collect::<Vec<_>>()
                    .join(",")
                    .as_ref(),
            ),
            Cell::new(&f.command.to_string()),
        ];
        table.add_row(Row::new(cells));
    }

    table.printstd();
}
