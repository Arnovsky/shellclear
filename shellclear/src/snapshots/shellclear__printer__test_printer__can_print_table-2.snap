---
source: shellclear/src/printer.rs
expression: "str::from_utf8(&out)"
---
Ok(
    "+---+-------+----------------------+--------------+\n| # | Shell | Name                 | Command      |\n+---+-------+----------------------+--------------+\n| 1 | Zshrc | test name,test name2 | test command |\n+---+-------+----------------------+--------------+\n",
)
